internal/teststep/config.go

```diff
@@ -21,7 +21,7 @@ const (
 )
 
 var (
-	providerConfigBlockRegex  = regexp.MustCompile(`provider "?[a-zA-Z0-9_-]+"? {`)
+	providerConfigBlockRegex  = regexp.MustCompile(`provider(\s*"[a-zA-Z0-9_-]+"\s*|\s+[a-zA-Z0-9_-]+\s*){`)
 	terraformConfigBlockRegex = regexp.MustCompile(`terraform {`)
```

Comment:
FYI: I presume the same whitespace problem can occur with `terraform{` as well, although `terraform validate` may not raise an error and Terraform may instead merge the `terraform` configuration block... configurations.

Reply:
Good call. I can verify that having duplicate `terraform` blocks can lead to issues. For example:

```shell
        Error: Duplicate required providers configuration
        
          on terraform_plugin_test.tf line 13, in terraform:
          13:   required_providers {
        
        A module may have only one required providers configuration. The required
        providers were previously configured at terraform_plugin_test.tf:3,3-21.
    testing_new.go:80: Error retrieving state, there may be dangling resources: exit status 1
```

Have modified the regex for detection of `terraform` block, and updated the tests.